### PR TITLE
C support

### PR DESCRIPTION
**Pull request by vext01**
Add's C support and tidies up build system (a lot, including making Java optional).

OK?

### CODE REVIEW

**Comment by vext01** [comment]
(Sorry this is such a large change)

File: iterations_runners/Makefile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+HERE != pwd
```
Comment (ltratt) [inline iterations_runners/Makefile]: Can we use a better variable name? Or, preferably, simpy use `pwd` instead of `${HERE}` later?

**Comment by ltratt** [comment]
This patch seems to use hard tabs everywhere, instead of 4 spaces. [Obviously in a Makefile, one has no choice. But everywhere else we should be using spaces.]

File: iterations_runners/Makefile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+C_ITER_RUNNER_CFLAGS = -fPIC -Wall -Wextra -pedantic -g
```
Comment (snim2) [inline iterations_runners/Makefile]: Compiling with `-g` probably isn't necessary for benchmarking.

**Comment by vext01** [comment]
The iterations runner is in KNF.

**Comment by ltratt** [comment]
Tabs are an editing disaster. Let's use 4 spaces, please.

File: iterations_runners/Makefile (hunk 2 above)
Comment (ltratt) [inline iterations_runners/Makefile]: Good spot. IIRC, it actually causes minor slowdown (bigger files etc.). I have no idea whether it's a measurable slowdown, though.

**Comment by vext01** [comment]
As you wish.

**Comment by vext01** [comment]
Try this.

If OK, will do some minimal squashing.

**Comment by ltratt** [comment]
Looks good to me!

**Comment by snim2** [comment]
I think this is still compiling the iter-runner with `-g`?

**Comment by vext01** [comment]
Nice catch. If the user wants symbols, they add `-g` to `CFLAGS`. E.g. `gmake CFLAGS=-g`.

**Comment by snim2** [comment]
OK, lgtm. Do you want to squash any commits before I merge?

**Comment by vext01** [comment]
Splat. I propose this commit structure.

**Comment by vext01** [comment]
Thanks all!

**Pull request merged**